.github/workflows/colcon.yml

```
@@ -0,0 +1,64 @@
+---
+name: colcon build/test
+
+on:
+  pull_request:
+  push:
+    branches: [humble]
+
+concurrency:
+  group: ci-${{github.workflow}}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-test:
+    runs-on: ubuntu-22.04
+    container: ardupilot/ardupilot-dev-ros:latest
+    strategy:
+      fail-fast: false  # don't cancel if a job from the matrix fails
+    steps:
+      # git checkout the PR
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+          path: src/ardupilot_ros
+      # https://ardupilot.org/dev/docs/ros2.html#installation-ubuntu
+      - name: Pull in repos with vcs
+        run: |
+          cd src
+          apt update && apt -y install wget python3-pip
+          wget --progress=dot:giga https://raw.githubusercontent.com/ArduPilot/ardupilot_gz/main/ros2_gz.repos
+          vcs import --recursive --debug --shallow --skip-existing < ros2_gz.repos
+      - name: Install rosdep dependencies
+        shell: 'bash'
+        run: |
+          apt update
+          apt install python3
```

```suggestion
          apt-get install -y python3
```

... similarly elsewhere.  Otherwise you will get warnings about `apt` not having a stable command-line